src/binary/coreir.cpp

```diff
@@ -19,12 +21,59 @@
 using namespace std;
 using namespace CoreIR;
 
+
 string getExt(string s) {
   auto split = splitString<vector<string>>(s,'.');
   ASSERT(split.size()>=2,"File needs extention: " + s);
   return split[split.size()-1];
 }
 
+bool fileExists(string name) {
+  std:: ifstream infile(name);
+  return infile.good();
+}
+
+void parseLib(string lib,string& libname,string& filename,vector<string>& lpaths) {
+  struct utsname unameData;
+  uname(&unameData);
+  string osname = unameData.sysname;
+  cout << "osname " << osname << endl;
+  string ext;
+  if (osname=="Darwin") {
+    ext = "dylib";
+  }
+  else if (osname=="Linux") {
+    ext = "so";
+  }
+  else {
+    ASSERT(0,"Cannot support OS " + osname);
+  }
+  
+  vector<string> f1parse = splitString<vector<string>>(lib,'/');
+  string libfile = f1parse[f1parse.size()-1];
+  vector<string> f2parse = splitString<vector<string>>(libfile,'.');
+  
+  if (f1parse.size()==1 && f2parse.size()==1) { //Just passed in the name of the library
+    libname = lib;
+    string filecheck = "libcoreir-" + libname + "." + ext;
+    for (auto path : lpaths) {
+      filename = path + "/" + filecheck;
+      if (fileExists(filename)) {
+        return;
+      }
+    }
+    ASSERT(0,"Cannot find lib " + lib);
```

Comment:
Could maybe print out the value of `lpaths` so the user knows where it looked (in case they forgot to add a path or added it to the wrong path)